src/mqt/bench/utils.py

```diff
@@ -313,38 +312,40 @@ def calc_qubit_index(qargs: list[Qubit], qregs: list[QuantumRegister], index: in
 def calc_supermarq_features(
     qc: QuantumCircuit,
 ) -> SupermarqFeatures:
-    qc = qc.remove_final_measurements(inplace=False)
-    qc = RemoveBarriers()(qc)
     connectivity_collection: list[list[int]] = []
     liveness_A_matrix = 0
     for _ in range(qc.num_qubits):
         connectivity_collection.append([])
 
-    for _, qargs, _ in qc.data:
-        liveness_A_matrix += len(qargs)
-        first_qubit = calc_qubit_index(qargs, qc.qregs, 0)
-        all_indices = [first_qubit]
-        if len(qargs) == 2:
-            second_qubit = calc_qubit_index(qargs, qc.qregs, 1)
-            all_indices.append(second_qubit)
-        for qubit_index in all_indices:
-            to_be_added_entries = all_indices.copy()
-            to_be_added_entries.remove(int(qubit_index))
-            connectivity_collection[int(qubit_index)].extend(to_be_added_entries)
+    for instruction, qargs, _ in qc.data:
+        if instruction.name != "barrier" and instruction.name != "measure":
```

Comment:
Small nit: I would prefer a 
```python
if instruction.name == "barrier" or instruction.name == "measure":
  continue
```
here to reduce the indentation of the rest of the code block.

Reply:
Implemented that.